File: python/26a_381A_SerejaAndDima.py
def main():
    n = int(input())
    cards = input().split()
    for i in range(n):
        cards[i] = int(cards[i])
    sereja, dima = 0, 0
    for i in range(n):
        if i % 2 == 0:
            if cards[0] > cards[-1]:
                sereja += cards[0]
                cards.pop(0)
            else:
                sereja += cards[-1]
                cards.pop(-1)
        else:
            if cards[0] > cards[-1]:
                dima += cards[0]
                cards.pop(0)
            else:
                dima += cards[-1]
                cards.pop(-1)
    print(sereja, dima)

if __name__ == "__main__":
    main()

File: cpp/26a_381A_SerejaAndDima.cpp
#include<bits/stdc++.h>

int main(){
    int n, temp, sereja = 0, dima = 0;
    std::vector<int> cards;
    std::cin>>n;
    for(int i = 0; i < n; i++){
        std::cin>>temp;
        cards.push_back(temp);
    }
    for(int i = 0; i < n; i++){
        if(i % 2 == 0){
            if(cards[0] > cards[cards.size() - 1]){
                sereja += cards[0];
                cards.erase(cards.begin(), cards.begin() + 1);
            }
            else {
                sereja += cards[cards.size() - 1];
                cards.erase(cards.end() - 1, cards.end());
            }
        }
        else{
            if(cards[0] > cards[cards.size() - 1]){
                dima += cards[0];
                cards.erase(cards.begin(), cards.begin() + 1);
            }
            else {
                dima += cards[cards.size() - 1];
                cards.erase(cards.end() - 1, cards.end());
            }
        }
    }
    std::cout<<sereja<<" "<<dima;
    return 0;
}